spec/post_index_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+RSpec.describe 'posts/index', type: :feature do
+  before(:each) do
+    @user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Developer',
+                        posts_counter: 0)
+    @first_post = Post.create(author: @user, title: 'Hello', text: 'This is my first post', comments_counter: 5,
+                              likes_counter: 0)
+    5.times do |i|
+      Comment.create(text: "This is comment ##{i}", user_id: @user.id, post_id: @first_post.id)
+    end
+    visit user_posts_path(@user.id, @first_post)
+  end
+
+  it 'displays post body' do
+    expect(page).to have_content('This is my first post')
+  end
+
+  it 'displays post title' do
+    expect(page).to have_content('Hello')
+  end
+
+  it 'displays first comment' do
+    expect(page).to have_content('This is comment #1')
+  end
+
+  it 'displays number of comments' do
+    expect(page).to have_content('Comments: 10')
+  end
+
+  it 'displays number of likes' do
+    expect(page).to have_content('Likes: 0')
+  end
+
+  it 'displays username' do
+    expect(page).to have_content(@user.name)
+  end
+
+  it 'displays number of posts' do
+    expect(page).to have_content('Number of posts: 1')
+  end
+
+  it 'displays a section for pagination' do
+    expect(page).to have_content('Pagination')
+  end
+
+  it 'displays user photo' do
+    expect(page).to have_css("img[src*='https://unsplash.com/photos/F_-0BxGuVvo']")
+  end
+
+  it 'Redirects to the post show page when you click on a post' do
+    visit user_post_path(user_id: @user.id, id: @first_post.id)
+    expect(page).to have_current_path(user_post_path(user_id: @user.id, id: @first_post.id))
+  end
```

Comment:
- [ ] As mentioned by the previous reviewer, as per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=When%20I%20click%20on%20a%20post%2C%20it%20redirects%20me%20to%20that%20post%27s%20show%20page.), The test is supposed to assert redirection to the post's show page `when the post is clicked`. not `when visiting the user_post_path` as you did. Please fix that.

[Hint] It's supposed to be like that:
```ruby
click_link '# put the post link here'
expect(page).to have_current_path(user_post_path(user_id: @user.id, id: @first_post.id))
```

Reply:
Hi @Omar-Muhamad thank you for this, I actually missed the clicking part. I will definitely add it.